flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPoolTest.java

```diff
@@ -453,7 +453,7 @@ public void testMaxBuffersPerChannelAndAvailability() throws Exception {
         final BufferBuilder bufferBuilder02 = localBufferPool.requestBufferBuilderBlocking(0);
         assertFalse(localBufferPool.getAvailableFuture().isDone());
 
-        assertNull(localBufferPool.requestBufferBuilder(0));
+        final BufferBuilder bufferBuilder03 = localBufferPool.requestBufferBuilderBlocking(0);
```

Comment:
Can you add comment explaining why pool is not available, but it should return a buffer + mention "Check FLINK-27522 for more information"?

Reply:
Hi @pnowojski , thanks for your review.

I will add these comments here, what do you think? Or should we add these to `LocalBufferPool#requestMemorySegment`?

```
        // When this channel reaches maxBuffersPerChannel, LocalBufferPool will be unavailable. But
        // when a memory segment is requested from LocalBufferPool and availableMemorySegments isn't
        // empty, maxBuffersPerChannel will be ignored and buffers will continue to be allocated.
        // Check FLINK-27522 for more information.
```